azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/task/CreateOrUpdateAppServicePlanTask.java

```diff
@@ -28,12 +29,23 @@
     @AzureOperation(name = "appservice|plan.create_update", params = {"this.config.servicePlanName()"}, type = AzureOperation.Type.SERVICE)
     public IAppServicePlan execute() {
         final AzureAppService az = Azure.az(AzureAppService.class).subscription(config.subscriptionId());
-        az.appServicePlan(config.servicePlanResourceGroup(), config.servicePlanName());
         final IAppServicePlan appServicePlan = az.appServicePlan(config.servicePlanResourceGroup(), config.servicePlanName());
         final String servicePlanName = config.servicePlanName();
         if (!appServicePlan.exists()) {
             AzureMessager.getMessager().info(String.format(CREATE_APP_SERVICE_PLAN, servicePlanName));
             AzureTelemetry.getActionContext().setProperty(CREATE_NEW_APP_SERVICE_PLAN, String.valueOf(true));
+            if (config.os() == null) {
+                throw new AzureToolkitRuntimeException("Missing required configuration for 'runtime.os'.");
+            }
+
+            if (config.region() == null) {
+                throw new AzureToolkitRuntimeException("Missing required configuration for 'region'.");
+            }
+
+            if (config.pricingTier() == null) {
+                throw new AzureToolkitRuntimeException("Missing required configuration for 'pricingTier'.");
+            }
```

Comment:
why do these case happen ?
I think we'd better to find out why they come out and prevent that (or provide default values) rather than show a error to user.

Reply:
It will be handled in future feature of advanced valuation, at here, now add the null check to avoid show npe to user.